checkov/yaml_doc/base_registry.py

```diff
@@ -225,6 +225,23 @@ def get_result_key(self, check: BaseCheck,
                        skip_info: _SkippedCheck) -> str:
         if "__startline__" and "__endline__" in entity_configuration:
             return f'{entity_type}.{entity_name}.{check.id}[{entity_configuration["__startline__"]}:{entity_configuration["__endline__"]}]'
+        if isinstance(entity_configuration, list):
+            start_line = None
+            end_line = None
+            for sub_conf in entity_configuration:
+                if "__startline__" and "__endline__" in sub_conf:
```

Comment:
just realizing, this only checks, if `"__endline__" in sub_conf` the left side of the and is always `True`, because it is a not empty string. Please create a a tuple `("__startline__", "__endline__") as a constant in or outside of the class and then use something like
```python
if all(key in sub_conf for key in START_ENDLINE_KEYS):
```
same for line 226